src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,179 @@
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: string[];
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { queryLanguages, attributesForPrediction = ['*'], nbHits = 1 } =
+      widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastAnswersResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRenderFn = debounce(renderFn, 200);
+
+    return {
+      $$type: 'ais.answers',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        // eslint-disable-next-line no-warning-comments
+        // FIXME: remove this customization once the engine accepts url encoded query params
+        if (instantSearchInstance.client.transporter) {
+          instantSearchInstance.client.transporter.userAgent.value =
+            'answers-test';
+        }
+        renderFn(
+          {
+            ...this.getWidgetRenderState(initOptions),
+            instantSearchInstance: initOptions.instantSearchInstance,
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const { state, instantSearchInstance } = renderOptions;
+        const { query, index } = state;
+        const answersIndex = instantSearchInstance.client!.initIndex!(index);
+        if (!answersIndex.findAnswers) {
+          throw new Error(withUsage('`algoliasearch` >= 4.8.0 required.'));
+        }
+        if (!query) {
+          // renders nothing with empty query
+          lastAnswersResult = {};
+          isLoading = false;
+          renderFn(
+            {
+              ...this.getWidgetRenderState(renderOptions),
+              instantSearchInstance: renderOptions.instantSearchInstance,
+            },
+            false
+          );
+          return;
+        }
+
+        // render the loader
+        isLoading = true;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance: renderOptions.instantSearchInstance,
+          },
+          false
+        );
+
+        // call /answers API
+        runConcurrentSafePromise(
+          answersIndex.findAnswers(query, queryLanguages, {
```

Comment:
why is it the render that's debounced, and not the findAnswers itself?

Reply:
When I received multiple responses in a short time window, the multiple renderings made it look flashy.

Reply:
I mean that not only you should debounce the rendering, but also the sending of the requests (since the request time can fluctuate)

Reply:
https://github.com/algolia/instantsearch.js/pull/4581/commits/f446f18d5669d5c99312a2a348a0046acd457612
here you go